### ISSUE TITLE
[Windows] Server initialized but client not detecting

### ISSUE THREAD

**Issue by bls1999** [8426THMY/PR1Server#1]
I have my config set to the default values.  When I open the exe for the server, it initializes perfectly; However, when I enter the configuration details (IP, port) into the SWF, I get an infinite "Connecting..." screen.

I've allowed the program through my firewall and unblocked the port.  I've also run the program as both a standard user and an administrator.

**Comment by Delphinoid** [comment]
You'll probably have to go into Flash Player's Global Security Settings panel and set the SWF as a trusted file. Flash is really finnicky with this sort of thing.

**Comment by geecaleb** [comment]
Would you be willing to do the same thing for Platform Racing 3?

**Comment by 8426THMY** [comment]
> Would you be willing to do the same thing for Platform Racing 3?

Sorry for the late reply, and thanks for your interest! I did start working on something similar for PR3 a while back, but I've since stopped working on it. You're welcome to take a look [here](https://github.com/8426THMY/PR3Server). Like the PR1 server, this repository includes a modded client to allow connections to arbitrary addresses.

I think another group has already made a functioning server, which you can check out [here](https://github.com/aromaa/PlatformRacing3). To my knowledge, they use a highly modified client with a lot of community-added content, and I suspect you may not be able to use it to connect to your own address. That being said, I'm sure it would be possible to modify their server to work with my client, though you will of course lose all of the custom content.

**Comment by geecaleb** [comment]
How would i go about adding their server to your client? Im not familiar with coding. Thank you so much for your response!

**Comment by geecaleb** [comment]
Do you think I could pay you to add it to your client?

**Comment by 8426THMY** [comment]
> How would i go about adding their server to your client? Im not familiar with coding. Thank you so much for your response!

I'm not sure if I'm the right person to ask, unfortunately. You'd probably have more luck asking them -- plus, it's entirely possible that I'm wrong, and that what they've written is exactly what you're looking for after all.

I appreciate the offer, but although I would like to work on this sort of thing again, I don't really have the time these days. Still, I wish you the best of luck!

**Comment by geecaleb** [comment]
Yeah i asked them but he didnt give much information. Just said that all he would need to do is write some extra code for it to work (Im thinking for his swf?). The source code has a local address on the github but when I tried compiling it in visual studio I was getting errors for some reason. Do you think I could get your discord or something so I can ask questions if i need to? Im going to see if i can get it to compile in visual studio- if i can then all i need to do is start the servers (if it compiles it like pr2 offline) and then use the connect to server on your modded swf. Also I tried logging in under your client. Do you know what the address is supposed to be? I tried putting https://myip:port after getting the server started but it still wouldnt let me log in

**Comment by 8426THMY** [comment]
> Do you think I could get your discord or something so I can ask questions if i need to?

Sure, my Discord username is 8426thmy.

> Also I tried logging in under your client. Do you know what the address is supposed to be? I tried putting https://myip:port after getting the server started but it still wouldnt let me log in

If you're connecting via the same system that's hosting the server, you'll probably want your local address, which should be 127.0.0.1. By default, the program should host on port 8426, but this can be configured.
